contracts/trade-shield-contract/src/tests/process_spot_order/process_order_limit_amount.rs

```diff
@@ -0,0 +1,492 @@
+            perpetual_enabled: None,
+            reward_enabled: None,
+            leverage_enabled: None,
+            limit_process_order: Some(3),
+        },
+        &[],
+    )
+    .unwrap();
+
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[0].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[1].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[2].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[3].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[4].order_id,
+        Status::Pending,
+    );
+
+    // Execute the order processing.
+    let resp = app.wasm_sudo(addr.clone(), &sudo_msg).unwrap();
+
+    assert_eq!(
+        app.wrap()
+            .query_balance(&addr, "btc")
+            .unwrap()
+            .amount
+            .u128(),
+        11
+    );
+    assert_eq!(
+        app.wrap()
+            .query_balance(&addr, "eth")
+            .unwrap()
+            .amount
+            .u128(),
+        9
+    );
+    assert_eq!(
+        app.wrap()
+            .query_balance(
+                &addr,
+                "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"
+            )
+            .unwrap()
+            .amount
+            .u128(),
+        0
+    );
+
+    assert_eq!(
+        app.wrap()
+            .query_balance("user", "btc")
+            .unwrap()
+            .amount
+            .u128(),
+        0
+    );
+    assert_eq!(
+        app.wrap()
+            .query_balance("user", "eth")
+            .unwrap()
+            .amount
+            .u128(),
+        0
+    );
+    assert_eq!(
+        app.wrap()
+            .query_balance(
+                "user",
+                "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"
+            )
+            .unwrap()
+            .amount
+            .u128(),
+        0
+    );
+
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[0].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[1].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[2].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[3].order_id,
+        Status::Pending,
+    );
+    test_spot_order_status(
+        &app.wrap(),
+        addr.to_string(),
+        spot_orders[4].order_id,
+        Status::Pending,
+    );
```

Comment:
Why do these orders are still pending after calling sudo_msg ?

Reply:
because here the price needed to process the order wasn't reach yet.